pkg/fioapp/layers.go

```diff
@@ -83,16 +79,49 @@ func GetAppLayersFromMap(ctx context.Context, svcImages map[string]string) (map[
 			return nil, err
 		}
 
-		manifestList, err := GetManifestList(ctx, imageManifestSvc, imageRef.(reference.Canonical).Digest())
+		indexManifest, err := imageManifestSvc.Get(ctx, imageRef.(reference.Canonical).Digest())
 		if err != nil {
 			return nil, err
 		}
 
-		for _, manifest := range manifestList.Manifests {
-			if _, ok := archToManifestList[manifest.Platform.Architecture]; !ok {
-				archToManifestList[manifest.Platform.Architecture] = make(map[distribution.ManifestService]digest.Digest)
+		switch im := indexManifest.(type) {
+		case *manifestlist.DeserializedManifestList:
+			{
+				manifestList := im
```

Comment:
this is an unusual way to write golang. you could delete this line and just do `im.Manifests` in the next line.

Reply:
Yes, but `go lint` is happy. Again, I find it more readable, in this case it's clear what exactly is `im`, is it a manifest list or a manifest. But, I am going to move the case code into dedicated functions so the additional variables won't be needed since `im` will be passed to them as parameter.